### PR TITLE
Tests for `quat_sandwich`

### PR DESCRIPTION
**Pull request by PNieck**
## Description

This PR contains test for `quat_sandwich` function. Moreover it fixes this function.

## Motivation and Context

More precise tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: algebra/tests/quat/various.c [hunk 1]
```diff
@@ -38,12 +38,19 @@ static const quat_t Q2 = { .a = 3.0f, .i = 0.0f, .j = 6.0f, .k = 1.0f };
...
 
 static const quat_t Q2timesQ3 = { .a = -8.0f, .i = 35.0f, .j = 11.0f, .k = 4.0f };
+static const quat_t Q2sandQ3 = { .a = 46.0f, .i = 118.0f, .j = 116.0f, .k = -190.0f };
 
 static const quat_t Q4 = { .a = 815.23f, .i = -818.07f, .j = -451.47f, .k = -546.79f };
 static const quat_t Q5 = { .a = 334.23f, .i = -822.81f, .j = 349.42f, .k = 548.18f };
 
 static const quat_t Q4timesQ5 = { .a = 56850.1358f, .i = -1000630.3952f, .j = 1032316.741f, .k = -393184.8904f };
-
+/* 613861170.772044 - 27264873.204711974i + 1092723638.076936j - 1585712993.277524k */
```
Comment (niewim19) [inline algebra/tests/quat/various.c:47]: This comment doubles the information from the quat below. Is it necessary?

File: algebra/tests/quat/various.c [hunk 2]
```diff
@@ -631,3 +638,48 @@ TEST_GROUP_RUNNER(group_quat_cjg)
...
+
+
+TEST_TEAR_DOWN(group_quat_sandwich)
+{
+}
+
+
+TEST(group_quat_sandwich, quat_sandwich_std)
+{
+	quat_t C;
```
Comment (niewim19) [inline algebra/tests/quat/various.c:663]: As discussed please use lowercase for local variables in testbody (with exception to matrices)

**Review by niewim19 — CHANGES_REQUESTED** [review]
Very good catch in quad_sandwitch implementation!

Only small remarks.

File: algebra/tests/quat/various.c (hunk 1 above)
Comment (PNieck) [reply algebra/tests/quat/various.c:47]: Sorry, my mistake - I have forgotten to delete it

**Review by niewim19 — APPROVED** [review]
LGTM

**Pull request merged**